statsmodels/sysreg/sysmodel.py

```diff
@@ -0,0 +1,622 @@
+
+        self.sigma = sigma
+        self.initialize()
+
+class SysOLS(SysWLS):
+    def __init__(self, sys, dfk=None, restrict_matrix=None, restrict_vect=None):
+        super(SysOLS, self).__init__(sys, weights=1.0, dfk=dfk,
+                restrict_matrix=restrict_matrix, restrict_vect=restrict_vect)
+
+class SysSUR(SysGLS):
+    def __init__(self, sys, dfk=None, restrict_matrix=None, restrict_vect=None):
+        super(SysSUR, self).__init__(sys, sigma=None,
+                restrict_matrix=restrict_matrix, restrict_vect=restrict_vect)
+
+        self.dfk = dfk
+
+        # Compute sigma by OLS equation by equation
+        resids = []
+        for eq in sys:
+            res = OLS(eq['endog'], eq['exog']).fit()
+            resids.append(res.resid)
+        resids = np.column_stack(resids)
+        sigma = self._compute_sigma(resids)
+
+        self.sigma = sigma
+        self.initialize()
+
+class SysSURI(SysOLS):
+    '''
+    SUR estimation with identical regressors in each equation.
+    It's equivalent to an OLS equation-by-equation.
+
+    Parameters
+    ----------
+    endogs : list of ndarray, (nobs,)
+        List of endogenous variable for each equation.
+    exog : ndarray, (nobs, k_vars)
+        Common exog variables in each equation.
+    '''
+    def __init__(self, endogs, exog):
+        # Build the corresponding system
+        sys = []
+        for endog in endogs:
+            eq = {}
+            eq['endog'] = endog
+            eq['exog'] = exog
+            sys.append(eq)
+        super(SysSURI, self).__init__(sys)
+
+class SysResults(LikelihoodModelResults):
+    '''
+    Attributes
+    ----------
+    cov_resids_est : ndarray
+        Residual covariance matrix used for estimation.
+    cov_resids : ndarray
+        Estimated residual covariance matrix with final residuals.
+    '''
+    def __init__(self, model, params, normalized_cov_params=None, scale=1.):
+        super(SysResults, self).__init__(model, params, normalized_cov_params,
+                scale)
+        self.cov_resids_est = model.sigma
+        # Compute sigma with final residuals
+        self.fittedvalues = model.predict(params=params, exog=None)
+        self.resids = model.endog - self.fittedvalues.reshape(model.neqs,-1).T
+        self.cov_resids = self._compute_cov_resids(self.resids)
+
+        self.nobs = model.nobs
+        self.df_resid = model.df_resid
+        self.df_model = model.df_model
+        self.ssr = np.array([sum(self.resids[:, eq] ** 2)
+            for eq in range(self.model.neqs)])
+        self.mse = self.ssr / self.df_resid
+        self.rmse = np.sqrt(self.mse)
+
+        means = np.mean(model.endog, axis=0)
+        self.sst = np.array([sum((model.endog[:,eq] - means[eq])**2)
+            for eq in range(self.model.neqs)])
+        self.rsquared = 1 - self.ssr / self.sst
+        self.rsquared_adj = 1 - (1 - self.rsquared)*(float(self.model.nobs - 1) /
```

Comment:
many of these should be cached attributes, or inherited from likelihood model results, or maybe better RegressionResults